### PR TITLE
fix: clarify some comments and rename some files in env

### PR DESCRIPTION
**Pull request by juliusmarminge**
# Clarifications regarding env

- [x] I reviewed linter warnings + errors, resolved formatting, types and other issues related to my work
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

We get a lot of questions regarding the typesafe env so this PR tries to clarify some of this, as well as making some personal enhancements.

- I renamed some files, the previous paths, e.g. `src/env/env-schema.mjs` had redundant double-env, so renamed to `src/env/schema.mjs`
- Made comments for the env-stuff ts-doc style, so they appear on hover.
- Clarified (imo) why you have to define your client-env twice.

---

## Additional notes

Should we prompt for typesafe env? Currently, the base project includes [the files by default](https://github.com/t3-oss/create-t3-app/tree/main/template/base/src/env) stuff for this and is not part of any installer. We had #196 that makes the env installer only go if either Prisma or NextAuth.js is selected, but the base stuff is still present regardless.

### CODE REVIEW

**Review by nexxeln — COMMENTED** [review]
Should we also add these things in the generated README?

**Comment by juliusmarminge** [comment]
> Should we also add these things in the generated README?

Yeah the t3-app/README can probably include some stuff for getting started and more "next steps" stuff

**Comment by juliusmarminge** [comment]
This is blocking #240 so merge when you looked it over. 

The README stuff should probably be a separate PR anyways.

File: template/base/src/env/schema.mjs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const clientSchema = z.object({
+  // NEXT_PUBLIC_BAR: z.string(),
+});
+
+/**
+ * You can't destruct `process.env` as a regular object, so you have to do
+ * it manually here. This is because Next.js evaluates this at build time,
+ * and only used environment variables are included in the build.
+ */
+export const clientEnv = {
```
Comment (corydeppen) [inline template/base/src/env/schema.mjs:26]: Since this isn't schema config, would it be appropriate to include it in `client.mjs` instead?

Comment (juliusmarminge) [reply template/base/src/env/schema.mjs:26]: Then you would have to change between files which seems messy. As it is now you dont even have to look st client/server.mjs. Atleast i prefer that

**Review by JacobMGEvans — APPROVED** [review]
I think these are great quality improvements!

**Comment by k4u5hik** [comment]
This is going to be very helpful for newbies like myself.

**Pull request merged**